src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractAISkeleton.java

```diff
@@ -27,21 +29,26 @@
 public abstract class AbstractAISkeleton<J extends AbstractJob> extends EntityAIBase
 {
 
-    private static final int                 MUTEX_MASK = 3;
+    private static final int                               MUTEX_MASK = 3;
     @NotNull
-    protected final      J                   job;
+    protected final      J                                 job;
     @NotNull
-    protected final      EntityCitizen       worker;
-    protected final      World               world;
+    protected final      EntityCitizen                     worker;
+    protected final      World                             world;
     @NotNull
-    protected final      ChatSpamFilter      chatSpamFilter;
+    protected final      ChatSpamFilter                    chatSpamFilter;
     @NotNull
-    private final        ArrayList<AITarget> targetList;
+    private final        Map<AIState, ArrayList<AITarget>> targetMap;
```

Comment:
I'm not yet convinced on any runtime benefits this will have with the amount of steps we currently have. Especially if we have to check each apply anyway. But I can understand that there is a reason to brewity of sorting them into different categories.

Reply:
The runtime benefit is that instead of doing this check: "if (state != target.getState() && target.getState() != null)" for all targets in the list we just check those which have the right state